src/materialized/src/http.rs

```diff
@@ -140,30 +141,44 @@ impl Server {
         };
 
         let svc = service::service_fn(move |req| {
-            let user = match user {
-                Ok(ref user) => user,
-                Err(e) => {
-                    return async move { Ok(util::error_response(StatusCode::UNAUTHORIZED, e)) }
-                        .boxed()
+            let user = user.clone();
+            let coord_client = self.coord_client.clone();
+            let start_time = self.start_time;
+            async move {
+                let user = match user {
+                    Ok(user) => user,
+                    Err(e) => return Ok(util::error_response(StatusCode::UNAUTHORIZED, e)),
+                };
+
+                let coord_client = coord_client.new_conn()?;
+                let session = Session::new(coord_client.conn_id(), user);
```

Comment:
If we are creating sessions in more than one place, I think doing the "session's drop impl checks that terminate was called" thing is almost certainly the best thing to do. Especially if we may also support TAIL and other stuff over http in the future.